add missing IDT2 font csp directives

rhenshaw56 opened the pull request:
Resolves #6611

**Overall change:**
_Adds missing font CSP directives._

**Code changes:**

- _add `news.files.bbci.co.uk` and `news.test.files.bbci.co.uk` to font-src CSP off follow up chat with @Tarqwyn ._
---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [X] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [X] This PR requires manual testing

rhenshaw56 committed:
add missing font csp directives

chris-hinds committed:
Merge branch 'latest' into idt2-font-src-csp

andrewscfc commented on src/server/utilities/cspHeader/index.js:
```diff
@@ -361,14 +361,27 @@ const directives = {
+    ampLive: [
+      'https://gel.files.bbci.co.uk', // Reith fonts
+      'https://ws-downloads.files.bbci.co.uk', // Other WS fonts
+      'news.files.bbci.co.uk', // STY Includes
```
shouldn't this have `https://` in, I think we want to restrict our csp to secure origins? We have the upgrade insecure flag set further down to upgrade any insecure origins used.

rhenshaw56 committed:
add secure flag to url

rhenshaw56 committed:
Merge branch 'idt2-font-src-csp' of https://github.com/bbc/simorgh into idt2-font-src-csp

chris-hinds committed:
Merge branch 'latest' into idt2-font-src-csp

paruchurisilpa committed:
Merge branch 'latest' into idt2-font-src-csp

paruchurisilpa committed:
Merge branch 'latest' into idt2-font-src-csp

paruchurisilpa commented:
I have tested this using:

* npm run build:live: debug && npm start 
* Go to http://localhost:7080/gahuza/amakuru-50332056?renderer_env=live

I still see ths font CSP errors.Attached is the screenshot..

Firefox:
<img width="1435" alt="Screenshot 2020-10-26 at 13 24 20" src="https://user-images.githubusercontent.com/1379277/97199111-8992ab00-17a7-11eb-916c-79e0706444f7.png"> 

Chrome:
![Uploading Screenshot 2020-10-26 at 16.21.01.png…]()

rhenshaw56 committed:
added self policy to font-src directive

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into idt2-font-src-csp

rhenshaw56 committed:
Merge branch 'idt2-font-src-csp' of https://github.com/bbc/simorgh into idt2-font-src-csp

andrewscfc commented:
I'm not sure we needed to make this change: https://github.com/bbc/simorgh/pull/8199/commits/6fd6ee15a87665afa4f3cf7af049e352f9a5d059

I've checked out the branch and if you look in the network tab, you see several font requests 404:
![image](https://user-images.githubusercontent.com/9645462/97276064-9f977e80-182e-11eb-8f7d-a88461041f3d.png)

And if you check these against a live url they also 404: http://www.bbc.co.uk/static/media/BBCReithSans_W_Rg.woff

I _think_ the origins you have added to csp are the locations the IDT2 team have presumably changed their includes to use in a new release of IDT2, is that what @Tarqwyn said in your conversation @rhenshaw56 ?

You may need to test this bug against a page that has a newly republished IDT2 include that had this problem, @lizcameron and @jimjohnsonrollings helped me with a similar issue last week.

I hope this makes sense @paruchurisilpa and @rhenshaw56 , happy to have a call if anything need clarifying?

rhenshaw56 commented:
> I'm not sure we needed to make this change: [6fd6ee1](https://github.com/bbc/simorgh/commit/6fd6ee15a87665afa4f3cf7af049e352f9a5d059)
> 
> I've checked out the branch and if you look in the network tab, you see several font requests 404:
> ![image](https://user-images.githubusercontent.com/9645462/97276064-9f977e80-182e-11eb-8f7d-a88461041f3d.png)
> 
> And if you check these against a live url they also 404: http://www.bbc.co.uk/static/media/BBCReithSans_W_Rg.woff
> 
> I _think_ the origins you have added to csp are the locations the IDT2 team have presumably changed their includes to use in a new release of IDT2, is that what @Tarqwyn said in your conversation @rhenshaw56 ?
> 
> You may need to test this bug against a page that has a newly republished IDT2 include that had this problem, @lizcameron and @jimjohnsonrollings helped me with a similar issue last week.
> 
> I hope this makes sense @paruchurisilpa and @rhenshaw56 , happy to have a call if anything need clarifying?

@andrewscfc thanks for having a look. @Tarqwyn's recommendation was adding the `news.files` ones I added earlier. The `'self'` I added was spotted during testing. I don't seem to remember exactly if that was a recurring issue when running the live build and isn't really needed, but will confirm.

andrewscfc commented:
Hey @rhenshaw56 , adding `self` gets rid of the csp error but then allows it to make a network request that 404s. I think with an up-to-date idt2, it would make requests to the new origins you added but I'm not sure?

rhenshaw56 commented:
> Hey @rhenshaw56 , adding `self` gets rid of the csp error but then allows it to make a network request that 404s. I think with an up-to-date idt2, it would make requests to the new origins you added but I'm not sure?

@andrewscfc you're right, those links don't exist, self policy just gets rid of the error (but I wonder why those links seem to be getting requested if they don't exist, maybe one for another issue)

rhenshaw56 commented:
@paruchurisilpa I just ran the local live build on the latest branch and I get the font csp violations you were seeing but none include specific 

![Screen Shot 2020-10-27 at 09 54 37](https://user-images.githubusercontent.com/25608386/97279292-1f2b4a80-183b-11eb-87d6-1fdfe082a0e0.png)

andrewscfc commented:
> > Hey @rhenshaw56 , adding `self` gets rid of the csp error but then allows it to make a network request that 404s. I think with an up-to-date idt2, it would make requests to the new origins you added but I'm not sure?
> 
> @andrewscfc you're right, those links don't exist, self policy just gets rid of the error (but I wonder why those links seem to be getting requested if they don't exist, maybe one for another issue)

I think the previous release of IDT2 includes make requests for these fonts, but the latest release doesn't. If we republish the IDT2 include for a particular test asset, we should expect the errors to go away.

rhenshaw56 committed:
removed self policy

paruchurisilpa commented:
You may need to test this bug against a page that has a newly republished IDT2 include that had this problem.
 @lizcameron and @jimjohnsonrollings helped me with a similar issue last week.

``` @lizcameron or @jimjohnsonrollings would you be able to help me with this please?```

lizcameron commented:
I've just republished all the IDT2 assets on this test STY https://www.test.bbc.com/gahuza/23313911

paruchurisilpa committed:
Merge branch 'latest' into idt2-font-src-csp

paruchurisilpa commented:
* npm run build:test:debug && npm start we get the live CSP instead of that from the test environment which is a bug with our env logic which is making it use the live csp for the test environment which is why we get those violations, that probably needs to be fixed on another PR but on live it works.
* After running the local build for test ie. npm run build && debug && npm start you need to use the modHeader extension to set this value bbc-origin: https://test.bbc.com
* I dont see CSP errors anymore .

LGTM..
